src/fft.jl

```diff
@@ -1,63 +1,121 @@
 import FFTW
 
+# returns the lengths of the bounding rectangle in reciprocal space
+# that encloses the sphere of radius Gmax
+function bounding_rectangle(lattice::AbstractMatrix{T}, Gmax; tol=1e-8) where {T}
+    # If |B G| ≤ Gmax, then
+    # |Gi| ≤ |e_i^T B^-1 B G| ≤ |B^-T e_i| Gmax = |A_i| Gmax
+    # with B the reciprocal lattice matrix and A_i the i-th column of the lattice matrix
+    Glims_ = [norm(lattice[:, i]) / 2T(π) * Gmax for i in 1:3]
+
+    # Round up, unless exactly zero (in which case keep it zero in
+    # order to just have one G vector for 1D or 2D systems)
+    Glims = [0, 0, 0]
+    for i = 1:3
+        if Glims_[i] != 0
+            Glims[i] = ceil(Int, Glims_[i] .- tol)
+        end
+    end
+    Glims
+end
+
+_smallprimes() = [2, 3, 5, 7]
```

Comment:
Why a function? Why not just `const SMALLPRIMES = ` why not keep it inside the function?

Reply:
Also I'm not too sure about 7. I excluded it because it does do considerably worse in performances, e.g.  http://www.fftw.org/speed/Pentium4-3.60GHz-icc/

Reply:
http://www.fftw.org/speed/Pentium4-3.60GHz-icc/RADPSDV683-icc.3d.drxx.np2.png is not bad

Reply:
It's used in two functions so can't leave it inside. const is OK, but I don't like shouting, so function is nice

Reply:
http://www.fftw.org/speed/Pentium4-3.60GHz-icc/RADPSDV683-icc.3d.scxx.np2.png and http://www.fftw.org/speed/Pentium4-3.60GHz-icc/RADPSDV683-icc.3d.dcxx.np2.png is and that's what we use.

Reply:
It's pretty noisy but there doesn't seem to be a clear pattern of slowdown for factors of 7, so I would include 7 in there also. I don't particularly mind either way.

Reply:
I do what ABINIT does, but if you think 7 does not hurt, then ok.